### DISCUSSION TITLE
Add osu!verification when signing in from a new device.

### DISCUSSION THREAD

**Discussion by htmlpaws** [ppy/osu#20589] [Ideas] [answered]
**_Hello there!_**

I recently got a new laptop, and after logging in, I realized there is currently no account verification feature like there is on osu!stable. I feel it should be added because without it, random people are able to log into an account that might not be theirs and engage in social functions (chat, score upload, etc.) without any proof it's actually said user, which can be used to target users.

I think additionally in the future there should be a built in GUI for the code which could allow for input to be easier on mobile for example, where sometimes opening another app can fully close out the game, along with not having to open a popup browser which is annoying in general. Another security thing that I wana put in an as idea if support for 2fa apps, as having to use email can be kinda annoying if you get the code lost in a pile of emails or shoved in a random folder.

Thanks again,
_@htmlpaws (Ethacc on osu.)_

**Comment by Walavouchey** [comment, marked as answer]
https://github.com/ppy/osu/issues/20590
